worker/etag.go

```diff
@@ -0,0 +1,83 @@
+package worker
```

Comment:
`NOTE` I added this file but will keep PR in `DRAFT` because I'm not sure if we should go ahead with etags the way we are using them right now